converter/internal/prometheusconvert/prometheusconvert.go

```diff
@@ -52,50 +52,41 @@ func Convert(in []byte) ([]byte, diag.Diagnostics) {
 	return prettyByte, diags
 }
 
-type prometheusBlocks struct {
-	discoveryBlocks             []*builder.Block
-	discoveryRelabelBlocks      []*builder.Block
-	prometheusScrapeBlocks      []*builder.Block
-	prometheusRelabelBlocks     []*builder.Block
-	prometheusRemoteWriteBlocks []*builder.Block
-}
-
-func newPrometheusBlocks() *prometheusBlocks {
-	return &prometheusBlocks{
-		discoveryBlocks:             []*builder.Block{},
-		discoveryRelabelBlocks:      []*builder.Block{},
-		prometheusScrapeBlocks:      []*builder.Block{},
-		prometheusRelabelBlocks:     []*builder.Block{},
-		prometheusRemoteWriteBlocks: []*builder.Block{},
-	}
-}
-
 // AppendAll analyzes the entire prometheus config in memory and transforms it
 // into Flow Arguments. It then appends each argument to the file builder.
 // Exports from other components are correctly referenced to build the Flow
 // pipeline.
-func AppendAll(f *builder.File, promConfig *prom_config.Config) diag.Diagnostics {
+func AppendAll(f *builder.File, promConfig *prom_config.Config, labelPrefix string) diag.Diagnostics {
 	var diags diag.Diagnostics
 	pb := newPrometheusBlocks()
-	remoteWriteExports := appendPrometheusRemoteWrite(pb, promConfig)
 
+	remoteWriteLabel := labelPrefix
+	if remoteWriteLabel == "" {
+		remoteWriteLabel = "default"
+	}
+	remoteWriteExports := appendPrometheusRemoteWrite(pb, promConfig.GlobalConfig, promConfig.RemoteWriteConfigs, remoteWriteLabel)
 	remoteWriteForwardTo := []storage.Appendable{remoteWriteExports.Receiver}
+
 	scrapeForwardTo := remoteWriteForwardTo
 	for _, scrapeConfig := range promConfig.ScrapeConfigs {
-		promMetricsRelabelExports := appendPrometheusRelabel(pb, scrapeConfig.MetricRelabelConfigs, remoteWriteForwardTo, scrapeConfig.JobName)
+		label := scrapeConfig.JobName
+		if labelPrefix != "" {
```

Comment:
Should this be remoteWriteLabel, or does default not apply here? Or to prevent confusion should the check for `remoteWriteLabel == ""` be embeded into the `appendPrometheusRetmoreWrite` func? I dont see it being used aside from the call to that func.